<h1 style="text-align:center; font-weight:bold; border: 2px solid black; padding: 10px; border-radius: 10px; background-color: #f2f2f2;">
    <span style="color: #66b3ff;">Know Your Health</span>: Predict Your Obesity Risk with Ease
</h1>


<center>
  <div style="border: 4px solid #66b3ff; padding: 10px; display: inline-block; background-color: #fff; border-radius: 12px; box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2); transform: scale(1); transition: transform 0.3s ease;">
    <img src="https://cdn.pixabay.com/photo/2022/03/07/07/20/weighing-scale-7053082_1280.jpg" width="600" style="border-radius: 8px;"/>
  </div>
</center>


<style>
    /* Table of contents container */
    .toc {
        background-color: #3e3e3e; /* Medium grey background */
        color: #fff;
        padding: 12px 18px;
        margin: 20px 0;
        border-radius: 12px;
        box-shadow: 0 6px 12px rgba(0, 0, 0, 0.2);
        width: 400px;
        font-size: 1em;
        font-family: 'Arial', sans-serif;
        border: 3px solid #5a5a5a; /* Darker grey border */
    }

    /* Header */
    .toc h1 {
        text-align: center;
        font-size: 1.8em;
        color: #f8f9fa;
        text-transform: uppercase;
        margin-bottom: 12px;
        font-family: 'Verdana', sans-serif;
        letter-spacing: 1px;
        text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.2);
    }

    /* Unordered list styling */
    .toc ul {
        list-style-type: none;
        padding-left: 0;
        margin: 0;
    }

    /* List item styling */
    .toc li {
        margin: 6px 0;
        font-size: 1.1em;
        text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.2);
    }

    /* Link styling */
    .toc a {
        color: #f1f1f1;
        text-decoration: none;
        padding: 6px 10px;
        border-radius: 6px;
        display: block;
        background-color: #545454; /* Grey tone */
        transition: background-color 0.3s, color 0.3s;
    }

    /* Hover effect for links */
    .toc a:hover {
        background-color: #6b6b6b; /* Lighter grey on hover */
        color: white;
        box-shadow: 0 0 8px rgba(255, 255, 255, 0.3);
    }

    /* Active state effect for links */
    .toc a:active {
        background-color: #444; /* Darker active state */
        color: white;
        box-shadow: 0 0 6px rgba(255, 255, 255, 0.5);
    }

    /* Nested list styling (subsections) */
    .toc .sublist {
        margin-left: 10px;
        list-style-type: circle;
        padding-left: 8px;
    }

    .toc .sublist a {
        background-color: #666; /* Slightly lighter grey */
        padding: 5px 10px;
        border-radius: 6px;
        transition: background-color 0.3s, color 0.3s;
    }

    .toc .sublist a:hover {
        background-color: #444; /* Darker grey hover */
        color: white;
        box-shadow: 0 0 6px rgba(255, 255, 255, 0.3);
    }

</style>

<div class="toc">
    <h1>Table of Contents</h1>
    <ul>
        <li><a href="#1-Introduction">1. Introduction</a>
            <ul class="sublist">
                <li><a href="#1-1-Problem-Statement">1.1. Problem Statement</a></li>
                <li><a href="#1-2-Objectives">1.2. Objectives</a></li>
            </ul>
        </li>
        <li><a href="#2-Importing-Packages">2. Importing Packages</a></li>
        <li><a href="#3-Data-loading">3. Data Loading</a></li>
        <li><a href="#4-Data-Cleaning">4. Data Cleaning</a></li>
        <li><a href="#5-Data-Preprocessing">5. Data Preprocessing</a></li>
        <li><a href="#6-Exploratory-Data-Analysis-EDA">6. Exploratory Data Analysis (EDA)</a></li>
        <li><a href="#7-Model-development">7. Model Development</a></li>
        <li><a href="#8-Model-Evaluation">8. Model Evaluation</a></li>
        <li><a href="#9-Model-Deployment">9. Model Deployment</a></li>
        <li><a href="#10-Conclusion">10. Conclusion</a></li>
        <li><a href="#11-Recommendations">11. Recommendations</a></li>
        <li><a href="#12-References">12. References</a></li>
    </ul>
</div>


#

<div style="text-align: center; border-top: 3px solid black; padding: 10px;">
    <h2><strong>1. Introduction</strong></h2>
</div>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <style>
        body { font-family: Arial, sans-serif; background: #f4f4f4; padding: 40px; }
        .container { max-width: 800px; margin: 0 auto; background: white; padding: 20px; border-radius: 8px; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1); }
        h2 { text-align: center; color: #333; }
        p { line-height: 1.6; color: #555; }
        .highlight { color: #66b3ff; font-weight: bold; } /* Lighter blue */
    </style>
</head>
<body>
    <div class="container">
        <p>This project aims to develop a predictive model that assesses the risk of <span class="highlight">obesity</span> based on a range of health and lifestyle factors. By analyzing key attributes like <span class="highlight">age</span>, <span class="highlight">height</span>, <span class="highlight">weight</span>, <span class="highlight">family history</span>, <span class="highlight">physical activity</span>, <span class="highlight">eating habits</span>, and more, the model will provide an accurate prediction of <span class="highlight">obesity</span> status. The goal is to help individuals understand their <span class="highlight">risk factors</span> and make informed decisions to improve their health and well-being. Using data science techniques like <span class="highlight">classification</span> and <span class="highlight">machine learning</span>, the project explores the relationship between lifestyle factors and obesity risk to create a reliable prediction tool.</p>
    </div>
</body>
</html>


<div style="text-align: center; border-top: 3px solid black; padding: 10px; width: 60%; margin: 0 auto;">
    <h3><strong>1.1. Problem Statement</strong></h3>
</div>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <style>
        body { font-family: Arial, sans-serif; background: #f4f4f4; padding: 40px; }
        .container { max-width: 800px; margin: 0 auto; background: white; padding: 20px; border-radius: 8px; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1); }
        h2 { text-align: center; color: #333; }
        p { line-height: 1.6; color: #555; }
        .highlight { color: #66b3ff; font-weight: bold; } /* Lighter blue */
    </style>
</head>
<body>
    <div class="container">
        <p><span class="highlight">Obesity</span> is a growing global health concern, influenced by <span class="highlight">genetic</span>, <span class="highlight">dietary</span>, and <span class="highlight">lifestyle factors</span>. Early prediction of <span class="highlight">obesity risk</span> can help in designing preventive measures and promoting healthier lifestyles. This project aims to analyze various factors contributing to <span class="highlight">obesity</span> and build <span class="highlight">machine learning models</span> to predict <span class="highlight">obesity levels</span> based on individual characteristics and behaviors.</p>
    </div>
</body>
</html>


<div style="text-align: center; border-top: 3px solid black; padding: 10px; width: 60%; margin: 0 auto;">
    <h3><strong>1.2. Objectives</strong></h3>
</div>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <style>
        body { font-family: Arial, sans-serif; background: #f4f4f4; padding: 40px; }
        .container { max-width: 800px; margin: 0 auto; background: white; padding: 20px; border-radius: 8px; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1); }
        h2 { text-align: center; color: #333; }
        p { line-height: 1.6; color: #555; }
        .highlight { color: #66b3ff; font-weight: bold; } /* Lighter blue */
    </style>
</head>
<body>
    <div class="container">
        <ul>
            <li><span class="highlight">Explore and preprocess data:</span> Clean, analyze, and visualize patterns.</li>
            <li><span class="highlight">Identify key contributors:</span> Assess lifestyle, diet, and activity impacts.</li>
            <li><span class="highlight">Develop predictive models:</span> Use <span class="highlight">ML</span> to classify obesity levels.</li>
            <li><span class="highlight">Evaluate performance:</span> Measure accuracy, precision, and recall.</li>
            <li><span class="highlight">Generate insights:</span> Provide findings for obesity prevention.</li>
        </ul>
    </div>
</body>
</html>


#

<div style="text-align: center; border-top: 3px solid black; padding: 10px;">
    <h2><strong>2.Importing Packages</strong></h2>
</div>

In [1]:
# Basic Packages
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")

#

<div style="text-align: center; border-top: 3px solid black; padding: 10px;">
    <h2><strong>3.Data Loading and Inspecting</strong></h2>
</div>

In this step, we begin by loading the dataset into the environment and performing an initial inspection to understand its structure and the nature of the data

In [2]:
# Load csv file into a DataFrame
df = pd.read_csv("Obesity prediction.csv")

# Overview of DataFrame
df.head()

Gender   Age  Height  Weight family_history FAVC  FCVC  NCP       CAEC  \
0  Female  21.0    1.62    64.0            yes   no   2.0  3.0  Sometimes   
1  Female  21.0    1.52    56.0            yes   no   3.0  3.0  Sometimes   
2    Male  23.0    1.80    77.0            yes   no   2.0  3.0  Sometimes   
3    Male  27.0    1.80    87.0             no   no   3.0  3.0  Sometimes   
4    Male  22.0    1.78    89.8             no   no   2.0  1.0  Sometimes   

  SMOKE  CH2O  SCC  FAF  TUE        CALC                 MTRANS  \
0    no   2.0   no  0.0  1.0          no  Public_Transportation   
1   yes   3.0  yes  3.0  0.0   Sometimes  Public_Transportation   
2    no   2.0   no  2.0  1.0  Frequently  Public_Transportation   
3    no   2.0   no  2.0  0.0  Frequently                Walking   
4    no   2.0   no  0.0  0.0   Sometimes  Public_Transportation   

               Obesity  
0        Normal_Weight  
1        Normal_Weight  
2        Normal_Weight  
3   Overweight_Level_I  
4  Overweight_Level_II

In [3]:
print(df.shape)

(2111, 17)


overview of the dataset's size, the presence of missing values, and the types of data each column holds.

In [4]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2111 entries, 0 to 2110
Data columns (total 17 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Gender          2111 non-null   object 
 1   Age             2111 non-null   float64
 2   Height          2111 non-null   float64
 3   Weight          2111 non-null   float64
 4   family_history  2111 non-null   object 
 5   FAVC            2111 non-null   object 
 6   FCVC            2111 non-null   float64
 7   NCP             2111 non-null   float64
 8   CAEC            2111 non-null   object 
 9   SMOKE           2111 non-null   object 
 10  CH2O            2111 non-null   float64
 11  SCC             2111 non-null   object 
 12  FAF             2111 non-null   float64
 13  TUE             2111 non-null   float64
 14  CALC            2111 non-null   object 
 15  MTRANS          2111 non-null   object 
 16  Obesity         2111 non-null   object 
dtypes: float64(8), object(9)
memory u

    The dataset contains 2111 entries and 17 columns, with both numerical (8 columns of type float64) and categorical (9 columns of type object) data. There are no missing values in any column


In [ ]:
# describing the data
df.describe()

Age       Height       Weight         FCVC          NCP  \
count  2087.000000  2087.000000  2087.000000  2087.000000  2087.000000   
mean     24.353090     1.702674    86.858730     2.421466     2.701179   
std       6.368801     0.093186    26.190847     0.534737     0.764614   
min      14.000000     1.450000    39.000000     1.000000     1.000000   
25%      19.915937     1.630178    66.000000     2.000000     2.697467   
50%      22.847618     1.701584    83.101100     2.396265     3.000000   
75%      26.000000     1.769491   108.015907     3.000000     3.000000   
max      61.000000     1.980000   173.000000     3.000000     4.000000   

              CH2O          FAF          TUE  
count  2087.000000  2087.000000  2087.000000  
mean      2.004749     1.012812     0.663035  
std       0.608284     0.853475     0.608153  
min       1.000000     0.000000     0.000000  
25%       1.590922     0.124505     0.000000  
50%       2.000000     1.000000     0.630866  
75%       2.466193     1.678102     1.000000  
max       3.000000     3.000000     2.000000

The values mostly make sense and are realistic, though there are a couple of values that might seem a bit out of the ordinary. For example, a weight as low as 39 kg is on the extreme side which might indicate outliers, but it’s not impossible, some people may fall into that range. The height of 1.45 m is also on the shorter end, but it's definitely possible, especially for certain individuals. While these values are rare, they could still reflect real-world cases.

<div style="text-align: center; border-top: 3px solid black; padding: 10px;">
    <h2><strong>4.Data Cleaning</strong></h2>
</div>

After using the info() and describe() functions for initial data inspection, we saw that there were no missing values. However, we will verify this further by explicitly checking for any missing values. Additionally, we will check for any duplicate rows in the dataset to ensure the data is clean.

In [7]:
# verify if theres no missing rows

no_missing_rows = df.isna().sum()

print(no_missing_rows)

Gender            0
Age               0
Height            0
Weight            0
family_history    0
FAVC              0
FCVC              0
NCP               0
CAEC              0
SMOKE             0
CH2O              0
SCC               0
FAF               0
TUE               0
CALC              0
MTRANS            0
Obesity           0
dtype: int64


In [6]:
# Number of duplicate rows
duplicate_count_before = df.duplicated().sum()

# print duplicate count
print('Duplicate row count :',duplicate_count_before)

# remove duplicate rows
df = df.drop_duplicates()

# Number of duplicate rows after deleting
duplicate_count_after = df.duplicated().sum()

# print duplicate count
print('Duplicate row count after dropping duplicates:',duplicate_count_after)

Duplicate row count : 24
Duplicate row count after dropping duplicates: 0


    Besides the dataset having duplicates, no missing values were found. After removing the duplicates, the dataset is now clean and free of any missing data and it is now ready for further analysis

<div style="text-align: center; border-top: 3px solid black; padding: 10px;">
    <h2><strong>4.Exploratoray Data Analysis</strong></h2>
</div>